src/app/common/TagList.vue

```diff
@@ -1,5 +1,11 @@
 <template>
-  <KTruncate>
+  <component
+    :is="props.shouldTruncate ? 'KTruncate' : 'div'"
+    :width="props.shouldTruncate ? 'auto' : undefined"
+    :class="{
+      'tag-list': !props.shouldTruncate,
```

Comment:
oooh look, would using the `AnonymousComponent` mean you don't need to remember to do any of this? Not totally sure, but if so that would be another +1 for `AnonymousComponent`

Reply:
Not really, no. We need the `width` conditional only when using KTruncate and we need the `tag-list` class only when _not_ using KTruncate. The latter we can’t even do with AnonymousComponent because we do need an actual element.

See also https://github.com/kumahq/kuma-gui/pull/1798#discussion_r1401841381.